enzyme/Enzyme/AdjointGenerator.h

```diff
@@ -270,10 +270,13 @@ class AdjointGenerator
         IRBuilder<> BuilderZ(newi);
         Value *newip = nullptr;
 
-        bool needShadow = is_value_needed_in_reverse<ValueType::ShadowPtr>(
-            TR, gutils, &I,
-            /*toplevel*/ Mode == DerivativeMode::ReverseModeCombined,
-            oldUnreachable);
+        bool needShadow =
+            Mode == DerivativeMode::ForwardMode
+                ? false
```

Comment:
I think the correct answer here is that this should be true if the loaded value itself is used as a pointer. I'm fine leaving this as is for now, but at least make a note here and try writing a test that exhibits this property.